moda-contracts/src/CatalogFactory.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+import "./interfaces/ICatalogFactory.sol";
+import "./interfaces/ICatalog.sol";
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+
+contract CatalogFactory is ICatalogFactory {
+    /// Storage Variables
+    address public immutable catalogBeacon;
+    address[] public catalogs;
+
+    /// Errors
+    error AddressCannotBeZero();
+
+    /// Constructor
+    constructor(address catalogBeacon_) {
+        if (catalogBeacon_ == address(0)) {
+            revert AddressCannotBeZero();
+        }
+        catalogBeacon = catalogBeacon_;
+    }
+
+    /// External Functions
+    /**
+     * @inheritdoc ICatalogFactory
+     */
+    function deployCatalog(
+        string calldata catalogName,
+        string calldata chainId,
+        string calldata catalogVersion,
+        address modaRegistry,
+        address splitsFactory,
+        address modaBeneficiary
+    ) external {
+        address catalog = address(new BeaconProxy(catalogBeacon, ""));
+        catalogs.push(catalog);
+        ICatalog(catalog).initialize(
+            catalogName,
+            chainId,
```

Comment:
We can get rid of `chainId` and use `block.chainid` inside of the Catalog to dynamically look it up.

Alternatively we can do this (bad): 

```sol
function getChainID() external view returns (uint256) {
    uint256 id;
    assembly {
        id := chainid()
    }
    return id;
}
```

Reply:
No need to use chainId. Anyone who needs to call that should know what chain they are on

Reply:
Ok i have removed chain id from the Catalog info, the only place it appears is when creating the track id